Analysis/Tasks/PWGHF/taskLc.cxx

```diff
@@ -70,29 +70,29 @@ struct TaskLc {
+        registry.get<TH1>(HIST("hmass"))->Fill(InvMassLcpKpi(candidate));
       }
       if (candidate.isSelLcpiKp() >= d_selectionFlagLc) {
-        registry.get<TH1>("hmass")->Fill(InvMassLcpiKp(candidate));
+        registry.get<TH1>(HIST("hmass"))->Fill(InvMassLcpiKp(candidate));
       }
-      registry.get<TH1>("hptcand")->Fill(candidate.pt());
-      registry.get<TH1>("hptprong0")->Fill(candidate.ptProng0());
-      registry.get<TH1>("hptprong1")->Fill(candidate.ptProng1());
-      registry.get<TH1>("hptprong2")->Fill(candidate.ptProng2());
-      registry.get<TH1>("hdeclength")->Fill(candidate.decayLength());
-      registry.get<TH1>("hd0Prong0")->Fill(candidate.impactParameter0());
-      registry.get<TH1>("hd0Prong1")->Fill(candidate.impactParameter1());
-      registry.get<TH1>("hd0Prong2")->Fill(candidate.impactParameter2());
-      registry.get<TH1>("hCt")->Fill(CtLc(candidate));
-      registry.get<TH1>("hCPA")->Fill(candidate.cpa());
-      registry.get<TH1>("hEta")->Fill(candidate.eta());
-      registry.get<TH1>("hselectionstatus")->Fill(candidate.isSelLcpKpi());
-      registry.get<TH1>("hselectionstatus")->Fill(candidate.isSelLcpiKp());
-      registry.get<TH1>("hImpParErr")->Fill(candidate.errorImpactParameter0());
-      registry.get<TH1>("hImpParErr")->Fill(candidate.errorImpactParameter1());
-      registry.get<TH1>("hImpParErr")->Fill(candidate.errorImpactParameter2());
-      registry.get<TH1>("hDecLenErr")->Fill(candidate.errorDecayLength());
-      registry.get<TH1>("hDecLenErr")->Fill(candidate.chi2PCA());
+      registry.get<TH1>(HIST("hptcand"))->Fill(candidate.pt());
+      registry.get<TH1>(HIST("hptprong0"))->Fill(candidate.ptProng0());
+      registry.get<TH1>(HIST("hptprong1"))->Fill(candidate.ptProng1());
+      registry.get<TH1>(HIST("hptprong2"))->Fill(candidate.ptProng2());
+      registry.get<TH1>(HIST("hdeclength"))->Fill(candidate.decayLength());
+      registry.get<TH1>(HIST("hd0Prong0"))->Fill(candidate.impactParameter0());
+      registry.get<TH1>(HIST("hd0Prong1"))->Fill(candidate.impactParameter1());
+      registry.get<TH1>(HIST("hd0Prong2"))->Fill(candidate.impactParameter2());
+      registry.get<TH1>(HIST("hCt"))->Fill(CtLc(candidate));
+      registry.get<TH1>(HIST("hCPA"))->Fill(candidate.cpa());
+      registry.get<TH1>(HIST("hEta"))->Fill(candidate.eta());
+      registry.get<TH1>(HIST("hselectionstatus"))->Fill(candidate.isSelLcpKpi());
+      registry.get<TH1>(HIST("hselectionstatus"))->Fill(candidate.isSelLcpiKp());
+      registry.get<TH1>(HIST("hImpParErr"))->Fill(candidate.errorImpactParameter0());
+      registry.get<TH1>(HIST("hImpParErr"))->Fill(candidate.errorImpactParameter1());
+      registry.get<TH1>(HIST("hImpParErr"))->Fill(candidate.errorImpactParameter2());
+      registry.get<TH1>(HIST("hDecLenErr"))->Fill(candidate.errorDecayLength());
+      registry.get<TH1>(HIST("hDecLenErr"))->Fill(candidate.chi2PCA());
     }
```

Comment:
You should use registry.fill instead...

Reply:
Hi @jgrosseo. I'm propagating the updates from the D0task that you can find here:

https://github.com/AliceO2Group/AliceO2/pull/5050/files#diff-7da89d2b691b60a9b20ec58616b3e46d2ba476b280c559bdc9dd4df035d83a39

Anyway, If we decide for a different strategy we have to propagate it to all the task.

Reply:
Yes, please use fill and not get unless you keep the pointer to the histogram. It should be straightforward to replace it. Let me know if you run into any trouble.

Reply:
Hi @jgrosseo . I modified the task following your indication. Let me know if it looks good to you. If so, I can take care of the propagation to the other tasks and push them before that the pull request is merged.
Thanks a lot for the review!

Reply:
Looks good. Thanks.